amd_openvx_extensions/amd_rpp/source/tensor/Normalize.cpp

```diff
@@ -77,10 +88,42 @@ static vx_status VX_CALLBACK refreshNormalize(vx_node node, const vx_reference *
             dst_roi[i].xywhROI.roiHeight = src_roi[i].xywhROI.roiHeight;
         }
         data->pSrcRoi = static_cast<unsigned *>(data->pSrcDims);
+    } else if (numDims == 4) {
+        RpptROI *src_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr);
+        RpptROI *dst_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr_dst);
+        for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+            unsigned index = i * 3 * 2;
+            if (data->inputLayout == vxTensorLayout::VX_NHWC) {
+                data->pSrcDims[index + 0] = src_roi[i].xywhROI.xy.y;
+                data->pSrcDims[index + 1] = src_roi[i].xywhROI.xy.x;
+                data->pSrcDims[index + 2] = 0;
+                data->pSrcDims[index + 3] = src_roi[i].xywhROI.roiHeight;
+                data->pSrcDims[index + 4] = src_roi[i].xywhROI.roiWidth;
+                data->pSrcDims[index + 5] = data->inputTensorDims[3];
+            } else if (data->inputLayout == vxTensorLayout::VX_NCHW) {
+                data->pSrcDims[index + 0] = 0;
+                data->pSrcDims[index + 1] = src_roi[i].xywhROI.xy.y;
+                data->pSrcDims[index + 2] = src_roi[i].xywhROI.xy.x;
+                data->pSrcDims[index + 3] = data->inputTensorDims[3];
+                data->pSrcDims[index + 4] = src_roi[i].xywhROI.roiHeight;
+                data->pSrcDims[index + 5] = src_roi[i].xywhROI.roiWidth;
+            }
+            dst_roi[i].xywhROI.xy.x = src_roi[i].xywhROI.xy.x;
+            dst_roi[i].xywhROI.xy.y = src_roi[i].xywhROI.xy.y;
+            dst_roi[i].xywhROI.roiWidth = src_roi[i].xywhROI.roiWidth;
+            dst_roi[i].xywhROI.roiHeight = src_roi[i].xywhROI.roiHeight;
+        }
+        data->pSrcRoi = static_cast<unsigned *>(data->pSrcDims);
     } else {
         data->pSrcRoi = static_cast<unsigned *>(roi_tensor_ptr);
+        Rpp32u *dst_roi = static_cast<unsigned *>(roi_tensor_ptr_dst);
+        for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+            unsigned index = i * nDim * 2;
+            for (Rpp32u j = 0; j < nDim; j++) {
+                dst_roi[index + j + nDim] = data->pSrcRoi[index + j + nDim];
```

Comment:
Why are we adding nDim here?

Reply:
This is the else case which supports 4D inputs so nDim is used here instead of 3